pkg/authorization/api/validation/validation.go

```diff
@@ -262,12 +266,24 @@ func validateRole(role *authorizationapi.Role, isNamespaced bool, fldPath *field
 }
 
 func ValidateRoleUpdate(role *authorizationapi.Role, oldRole *authorizationapi.Role, isNamespaced bool) field.ErrorList {
-	allErrs := ValidateRole(role, isNamespaced)
+	allErrs := validateRoleMeta(role, isNamespaced, nil)
 	allErrs = append(allErrs, validation.ValidateObjectMetaUpdate(&role.ObjectMeta, &oldRole.ObjectMeta, field.NewPath("metadata"))...)
-
+	if countAttributeRestrictions(role) > countAttributeRestrictions(oldRole) {
+		allErrs = append(allErrs, field.Invalid(field.NewPath("rules"), role.Rules, "must contain the same or less rules with attributeRestrictions"))
+	}
 	return allErrs
 }
 
+func countAttributeRestrictions(role *authorizationapi.Role) int {
```

Comment:
If a new role has a rule has an attribute restriction, make sure the exactly same rule (semantic.DeepEqual) exists in the old.